api/app/Console/Commands/CreateDigitalCareersPoolsCandidates.php

```diff
@@ -0,0 +1,214 @@
+        $candidateCollection = PoolCandidate::where('pool_id', $oldDigitalCareersId)->get();
+        $ITLevels = [1, 2, 3, 4];
+        $poolStreams = ApiEnums::poolStreams();
+        $dateNow = Carbon::now();
+        $expiryDate = Carbon::now()->addYear();
+
+        // organizing assets and streams mapping
+        $assetsCollection = CmoAsset::all();
+        global $assetArray;
+        $assetArray = [];
+        // index is asset model id field, value is asset key field
+        foreach ($assetsCollection as $asset) {
+            $assetArray[$asset->id] = $asset->key;
+        }
+        global $assetKeyToStreamMap;
+        $assetKeyToStreamMap = [
+            "db_admin" => ApiEnums::POOL_STREAM_DATABASE_MANAGEMENT,
+            "infrastructure_ops" => ApiEnums::POOL_STREAM_INFRASTRUCTURE_OPERATIONS,
+            "cybersecurity" => ApiEnums::POOL_STREAM_SECURITY,
+            "app_dev" => ApiEnums::POOL_STREAM_SOFTWARE_SOLUTIONS,
+            "enterprise_architecture" => ApiEnums::POOL_STREAM_ENTERPRISE_ARCHITECTURE,
+            // IT Business Analyst / IT Project Management -> Project Portfolio Management
+            //  ? -> Business Line Advisory Services
+            // ? -> Planning and Reporting
+            // Application Testing / Quality Assurance -> ?
+            // Data Science / Analysis -> ?
+            // Information Management (IM) -> ?
+        ];
+
+        /**
+         * check whether a candidate possesses an expected classification or fits by desired salary
+         * @return boolean
+        */
+        function candidateMatchesClassification(string $candidateId,
+                                                string $classificationId,
+                                                ?array $expectedSalary, // can be null
+                                                string $classificationMinSalary,
+                                                string $classificationMaxSalary): bool {
+
+            $candidateClassificationsArray = PoolCandidate::where('id', $candidateId)
+                                                ->first()
+                                                ->expectedClassifications()
+                                                ->pluck('classifications.id')
+                                                ->toArray();
+            if (in_array($classificationId, $candidateClassificationsArray)) {
+                return true;
+            }
+            if ($expectedSalary) {
+                $minSalaryInt = intval($classificationMinSalary);
+                $maxSalaryInt = intval($classificationMaxSalary);
+                $salaryArray = [];
+                // $expectedSalary is an array of strings, need to convert to array of array pairs, each pair creating lower/upper limits of ints
+                foreach ($expectedSalary as $salary) {
+                    if ($salary == '_50_59K'){
```

Comment:
Big if/else trees like this should probably be a switch instead.  https://www.php.net/manual/en/control-structures.switch.php

Reply:
switch block switched in here https://github.com/GCTC-NTGC/gc-digital-talent/pull/4467/commits/fc80a132fed5eb8aeaf60d8817dfbee4ad2e0437